### PR TITLE
RYA-372 Excluded org.json:json for incompatible license

### PR DESCRIPTION
**Pull request by DLotts**
## Description
Excluded org.json:json for incompatible license.  Apparently, the Giraph library depends on it, but does not need it.  

### Tests
no new tests, old ones serve

### Links
[Jira](https://issues.apache.org/jira/browse/RYA-372)

### Checklist
- [ ] Code Review
- [X] Squash Commits

#### People To Reivew
anyone

### CODE REVIEW

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-rya-master-with-optionals-pull-requests/484/<h2>Build result: FAILURE</span></h2>[...truncated 47.36 KB...][INFO] --- maven-source-plugin:2.4:jar-no-fork (attach-sources) @ rya.dao ---[INFO] [INFO] --- apache-rat-plugin:0.11:check (check-licenses) @ rya.dao ---[INFO] 53 implicit excludes (use -debug for more details).[INFO] Exclude: DEPENDENCIES[INFO] 1 resources included (use -debug for more details)[INFO] Rat check: Summary of files. Unapproved: 0 unknown: 0 generated: 0 approved: 1 licence.[INFO] [INFO] --- maven-enforcer-plugin:1.3.1:enforce (enforce-mvn) @ rya.dao ---[INFO] [INFO] --- maven-remote-resources-plugin:1.5:process (default) @ rya.dao ---[INFO] [INFO] --- animal-sniffer-maven-plugin:1.15:check (default) @ rya.dao ---[INFO] Checking unresolved references to org.codehaus.mojo.signature:java18:1.0Java HotSpot(TM) 64-Bit Server VM warning: INFO: os::commit_memory(0x0000000788080000, 938999808, 0) failed; error='Cannot allocate memory' (errno=12)## There is insufficient memory for the Java Runtime Environment to continue.# Native memory allocation (mmap) failed to map 938999808 bytes for committing reserved memory.# An error report file with more information is saved as:# /home/jenkins/jenkins-slave/workspace/incubator-rya-master-with-optionals-pull-requests/hs_err_pid30536.logERROR: Maven JVM terminated unexpectedly with exit code 1Setting status of 66faa4b8958a9de7f3fa35aa55c891ae38734885 to FAILURE with url https://builds.apache.org/job/incubator-rya-master-with-optionals-pull-requests/484/ and message: 'FAILURE 'Using context: Jenkins: clean package -Pgeoindexing

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-rya-master-with-optionals-pull-requests/483/

**Comment by DLotts** [comment]
asfbot build

**Comment by jdasch** [comment]
asfbot build

**Comment by jdasch** [comment]
asfbot build

**Comment by jdasch** [comment]
asfbot build

**Comment by jdasch** [comment]
asfbot build

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-rya-master-with-optionals-pull-requests/487/<h2>Failed Tests: <span class='status-failure'>3</span></h2><h3><a name='incubator-rya-master-with-optionals-pull-requests/org.apache.rya:rya.prospector' /><a href='https://builds.apache.org/job/incubator-rya-master-with-optionals-pull-requests/487/org.apache.rya$rya.prospector/testReport'>incubator-rya-master-with-optionals-pull-requests/org.apache.rya:rya.prospector</a>: <span class='status-failure'>3</span></h3><ul><li><a href='https://builds.apache.org/job/incubator-rya-master-with-optionals-pull-requests/487/org.apache.rya$rya.prospector/testReport/org.apache.rya.prospector.mr/ProspectorTest/testCount/'><strong>org.apache.rya.prospector.mr.ProspectorTest.testCount</strong></a></li><li><a href='https://builds.apache.org/job/incubator-rya-master-with-optionals-pull-requests/487/org.apache.rya$rya.prospector/testReport/org.apache.rya.prospector.service/ProspectorServiceEvalStatsDAOTest/testCount/'><strong>org.apache.rya.prospector.service.ProspectorServiceEvalStatsDAOTest.testCount</strong></a></li><li><a href='https://builds.apache.org/job/incubator-rya-master-with-optionals-pull-requests/487/org.apache.rya$rya.prospector/testReport/org.apache.rya.prospector.service/ProspectorServiceEvalStatsDAOTest/testNoAuthsCount/'><strong>org.apache.rya.prospector.service.ProspectorServiceEvalStatsDAOTest.testNoAuthsCount</strong></a></li></ul>

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-rya-master-with-optionals-pull-requests/488/

File: extras/rya.giraph/pom.xml [hunk 1]
```diff
@@ -35,6 +35,13 @@ under the License.
             <groupId>org.apache.giraph</groupId>
             <artifactId>giraph-core</artifactId>
             <version>1.2.0</version>
+            <exclusions>
+            	<!-- exclude for incompatible license -->
+                <exclusion>
+                    <artifactId>json</artifactId>
+                    <groupId>org.json</groupId>
+                </exclusion>
+            </exclusions>
```
Comment (jdasch) [inline extras/rya.giraph/pom.xml]: This change should be sufficient:
```
    <dependencies>
<!--         
        Uncomment the following drop-in replacement if the exclusion
        of the dependency org.json:json causes issues at runtime.
        <dependency>
            <groupId>com.tdunning</groupId>
            <artifactId>json</artifactId>
            <version>1.8</version>
            <scope>runtime</scope>
        </dependency>
 -->       
        <dependency>
            <groupId>org.apache.giraph</groupId>
            <artifactId>giraph-core</artifactId>
            <version>1.2.0</version>
            <exclusions>
                <!-- org.json:json excluded for incompatible license -->
                <exclusion>
                    <groupId>org.json</groupId>
                    <artifactId>json</artifactId>
                </exclusion>
            </exclusions>
        </dependency>
        <dependency>
            <groupId>org.apache.giraph</groupId>
            <artifactId>giraph-accumulo</artifactId>
            <version>1.2.0</version>
            <exclusions>
                <!-- org.json:json excluded for incompatible license -->
                <exclusion>
                    <groupId>org.json</groupId>
                    <artifactId>json</artifactId>
                </exclusion>
            </exclusions>
        </dependency>
```

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-rya-master-with-optionals-pull-requests/490/<h2>Build result: FAILURE</span></h2>[...truncated 1.34 MB...][INFO] Apache Rya Spark Support ........................... SKIPPED[INFO] Apache Rya Web Projects ............................ SKIPPED[INFO] Apache Rya Web Implementation ...................... SKIPPED[INFO] ------------------------------------------------------------------------[INFO] BUILD FAILURE[INFO] ------------------------------------------------------------------------[INFO] Total time: 09:33 min[INFO] Finished at: 2017-09-18T20:23:02+00:00[INFO] Final Memory: 288M/3004M[INFO] ------------------------------------------------------------------------[ERROR] Failed to execute goal org.apache.maven.plugins:maven-jar-plugin:2.5:jar (default-jar) on project rya.periodic.notification.service: Error assembling JAR: Failed to create directory: /home/jenkins/jenkins-slave/workspace/incubator-rya-master-with-optionals-pull-requests/extras/periodic.notification/service/target/maven-archiver -> [Help 1][ERROR] [ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.[ERROR] Re-run Maven using the -X switch to enable full debug logging.[ERROR] [ERROR] For more information about the errors and possible solutions, please read the following articles:[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException[ERROR] [ERROR] After correcting the problems, you can resume the build with the command[ERROR]   mvn <goals> -rf :rya.periodic.notification.servicechannel stoppedSetting status of 09c696079b14c1b4db5a84e7a6514c5010493153 to FAILURE with url https://builds.apache.org/job/incubator-rya-master-with-optionals-pull-requests/490/ and message: 'FAILURE 'Using context: Jenkins: clean package -Pgeoindexing

**Comment by DLotts** [comment]
asfbot build

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-rya-master-with-optionals-pull-requests/493/

**Pull request closed without merging**